autogluon/task/image_classification/classifier.py

```diff
@@ -112,37 +114,83 @@ def predict(self, X, input_size=224, plot=True):
         >>> test_data = task.Dataset('~/data/test', train=False)
         >>> class_index, class_probability = classifier.predict('example.jpg')
         """
-        # model inference
+
         input_size = self.model.input_size if hasattr(self.model, 'input_size') else input_size
-        resize = int(math.ceil(input_size / 0.875))
-        transform_fn = Compose([
-                Resize(resize),
-                CenterCrop(input_size),
-                ToTensor(),
-                transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
-            ])
-        def predict_img(img):
+        resize = int(math.ceil(input_size / resize_ratio))
+
+        transform_size = transforms.Compose([
+            transforms.Resize(resize),
+            transforms.CenterCrop(input_size),
+            transforms.ToTensor(),
+            transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+        ])
+
+        def predict_img(img, ensemble=False):
             proba = self.predict_proba(img)
-            ind = mx.nd.argmax(proba, axis=1).astype('int')
-            idx = mx.nd.stack(mx.nd.arange(proba.shape[0], ctx=proba.context), ind.astype('float32'))
-            probai = mx.nd.gather_nd(proba, idx)
-            return ind, probai, proba
+            if ensemble:
+                return proba
+            else:
+                ind = mx.nd.argmax(proba, axis=1).astype('int')
+                idx = mx.nd.stack(mx.nd.arange(proba.shape[0], ctx=proba.context), ind.astype('float32'))
+                probai = mx.nd.gather_nd(proba, idx)
+                return ind, probai, proba
+
+        def avg_prediction(different_dataset, threshold=0.001):
```

Comment:
nontrivial function with docstring missing, could you please add to this and similar?